crates/miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -1196,11 +1216,12 @@ export.was_procedure_called
 
     dup movdn.5
 
-    # get procedure info (PROC_ROOT, storage_offset, storage_size) from memory stored at index
-    exec.get_procedure_info
-    # => [MEM_PROC_ROOT, storage_offset, storage_size, PROC_ROOT, index]
+    # get procedure root from memory stored at index
+    exec.get_procedure_root
+    # => [MEM_PROC_ROOT, PROC_ROOT, index]
```

Comment:
This index which is passed to `get_procedure_root` and `get_acct_procedures_call_tracking_ptr` is coming from the advice provider, so I believe a malicious transaction host could arbitrarily offset into memory and affect the outcome of `was_procedure_called` and authentication as a result. It would certainly be difficult, but not impossible either. So I think we should validate this index similar to `get_procedure_info`, ensuring it is less than the `get_num_account_procedures`. I'd add this to `get_procedure_root` directly.

It might also be worth pointing out that we don't have to check the index for word-alignment because we multiply it by 8 in `export.get_acct_procedure_ptr`.

Maybe we should also add a "index assumed valid" note on `set_was_procedure_called`.